linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env groovy
+
+def PowerShellWrapper(psCmd) {
+    psCmd = psCmd.replaceAll("\r", "").replaceAll("\n", "")
+    def exitCode = bat (returnStatus: true, script: "powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \"\$ErrorActionPreference='Stop';[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;$psCmd;EXIT \$global:LastExitCode\"")
+    return exitCode
+}
+
+properties ([
+    parameters([
+        choice (name: 'Distro', choices: 'default\nxenial\nbionic\nboth', 
+                description: 'xenial - only Xenial validation <br> bionic - only Bionic validation <br> both - both distros validation <br>'),
+        choice (name: 'Kernel', choices: 'linux-azure\nlinux-azure-edge\nboth',
+                description: 'linux-azure - only linux-azure kernel validation <br> linux-azure-edge - only linux-azure-edge kernel validation <br> both - both kernels validation <br>'),
+        choice (name: 'Platform', choices: 'both\nhyperv\nazure',
+                description: 'hyperv - only Hyper-V validation <br> azure - only Azure validation <br> both - both platforms validation <br>')
+    ]),
+    pipelineTriggers([cron('H */12 * * *')])
+])
+
+def distroToTest = []
+DISTRO_TRIGGER = "default"
+DISTRO_BIONIC = "bionic"
+DISTRO_XENIAL = "xenial"
+DISTRO_BIONIC_EDGE = "bionic-edge"
+DISTRO_XENIAL_EDGE = "xenial-edge"
+DISTRO_BOTH = "both"
+KERNEL_AZURE = "linux-azure"
+KERNEL_AZURE_EDGE = "linux-azure-edge"
+KERNEL_BOTH = "both"
+PLATFORM_HYPERV = "hyperv"
+PLATFORM_AZURE = "azure"
+PLATFORM_BOTH = "both"
+HYPERV_PARAMS_AZURE = "azure_kernel=yes; azure_kernel_edge=no"
+HYPERV_PARAMS_AZURE_EDGE = "azure_kernel=no; azure_kernel_edge=yes"
+XENIAL_VERSION = "ubuntu_16.04.3"
+BIONIC_VERSION = "Ubuntu_18.04"
+AZURE_AUTOMATION_GIT = "https://github.com/iamshital/azure-linux-automation.git"
+ARM_XENIAL = "Canonical UbuntuServer 16.04-LTS latest"
+ARM_BIONIC = "Canonical UbuntuServer 18.04-LTS latest"
+EXIT_CODE_BOTH = "10"
+EXIT_CODE_XENIAL = "11"
+EXIT_CODE_BIONIC = "12"
+
+if ("${env:distro}" == "${DISTRO_TRIGGER}") {
+    stage ("Check for kernel update") {
+        node ("ws2016"){
+            withCredentials(bindings: [string(credentialsId: 'AZURE_KERNEL_INFO_LOCATION', variable: 'AZURE_KERNEL_INFO_LOCATION')]) {
+                dir (env.BUILD_NUMBER + env.BRANCH_NAME) { 
+                    checkout scm
+                    echo 'Checking if there is a new kernel version'
+                    kernel_sts = PowerShellWrapper('''
```

Comment:
rename the variable to something relevant like:
kernel_check_exit_code

in Groovy, the variable names are in camelCase format, not camel_case (rename all of them)